rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -560,6 +569,14 @@ public void internalAddNodeToCore(RMNode configuredNode) {
             return;
         }
 
+        //noinspection ConstantConditions
+        if(isNumberOfNodesLimited() && isMaximumNumberOfNodesReached()) {
```

Comment:
It would be nice to add unit tests in order to check your assertions and ensure that an _AddingNodesException_ is raised or not depending of the situation. Using Mockito, it should be easy.